pkg/kubelet/nodelease/controller.go

```diff
@@ -123,10 +116,11 @@ func (c *controller) backoffEnsureLease() (*coordv1beta1.Lease, bool) {
 // ensureLease creates the lease if it does not exist. Returns the lease and
 // a bool (true if this call created the lease), or any error that occurs.
 func (c *controller) ensureLease() (*coordv1beta1.Lease, bool, error) {
-	lease, err := c.leaseClient.Get(c.holderIdentity, metav1.GetOptions{})
+	leaseClient := c.client.CoordinationV1beta1().Leases(corev1.NamespaceNodeLease)
```

Comment:
understand the intention is to avoid all node lease OPs come to the same Api Server instance in the api-server- partitioned-clusters, but is this code ( line 119 ) will randomly pick up one Api Server ?

Reply:
Yes it does. The reason leaseClient could not do it is because it was initialized at kubelet startup and won't be able to pick up new api server(s) changes. c.client.CoordinationV1beta1().Leases(ns) returns the standard clientset, which will auotomatically pick up api server updates and randomize the target api server when request is sent. Promethus metrics apiserver_request_count{resource="leases"} proved node lease GET/PUT request from kubelet is now balanced across all api servers.